src/MPDevice.cpp

```diff
@@ -4385,45 +4422,74 @@ void MPDevice::setDataNode(QString service, const QByteArray &nodeData,
     QByteArray sdata = pMesProt->toByteArray(service);
     sdata.append((char)0);
 
-    jobs->append(new MPCommandJob(this, MPCmd::SET_DATA_SERVICE,
-                                  sdata,
-                                  [this, jobs, service](const QByteArray &data, bool &) -> bool
+    if (isBLE())
     {
-        if (pMesProt->getFirstPayloadByte(data) != 1)
+        sdata.append((char)0);
+        jobs->append(new MPCommandJob(this, MPCmd::ADD_DATA_SERVICE,
+                                      sdata,
+                                      [this, service, cb](const QByteArray &data, bool &) -> bool
         {
-            qWarning() << "context " << service << " does not exist";
-            //Context does not exists, create it
-            createJobAddContext(service, jobs, true);
-        }
-        else
-            qDebug() << "set_data_context " << service;
-        return true;
-    }));
+            if (pMesProt->getFirstPayloadByte(data) != 1)
```

Comment:
here as well

Reply:
MPDevice contains a lot of these checks, so I added to the todo list to fix the others too.